### PR TITLE
Fix subdomain fetching for subdomainless urls fixes #6710

### LINKED ISSUE #6710: Fix http authentication autocreation with organizations in subdomainless URLs
Testing Kerberos integration @danicarrion found out that users are not being created inside organizations for subdomainless URLs.

The patch is replacing

```
subdomain = CartoDB.subdomain_from_request(request)
```

with

```
subdomain = CartoDB.subdomainless_urls? ? request.host.to_s.gsub(".#{CartoDB.session_domain}", '') : CartoDB.subdomain_from_request(request)`
```

in `app/controllers/signup_controller.rb`.

### PR DESCRIPTION
**Pull request by juanignaciosl**
This fixes #6710. @javisantana CR this, please.

### CODE REVIEW

**Comment by javisantana** [comment]
:es: 

could you do the PR to 3.12.X so I tag it?

**Comment by juanignaciosl** [comment]
Done: CartoDB/cartodb#6717

**Comment by juanignaciosl** [comment]
@gfiorav , @javisantana , another CR, please. Organization signup wasn't working for subdomainless urls (#6724). I've also added it to 3.12.X branch PR (CartoDB/cartodb#6717).

File: app/controllers/signup_controller.rb [hunk 1]
```diff
@@ -130,7 +130,7 @@ def initialize_google_plus_config
   end
 
   def load_organization
-    subdomain = CartoDB.subdomain_from_request(request)
+    subdomain = CartoDB.subdomainless_urls? ? request.host.to_s.gsub(".#{CartoDB.session_domain}", '') : CartoDB.subdomain_from_request(request)
```
Comment (gfiorav) [inline app/controllers/signup_controller.rb]: Could `CartoDB.subdomain_from_request` automagically take care of subdomainless? I mean, check internally and give the proper subdomain back.

Comment (juanignaciosl) [inline app/controllers/signup_controller.rb]: It "does", but the default behaviour (returning an empty string) is not valid in order to load the organization.

**Comment by gfiorav** [comment]
Ok, :+1: !

**Pull request merged**